### PR TITLE
Add support for Feign.SetterFactory.

### PR DESCRIPTION
**Pull request by ErikKringen**
Fixes gh-1414

### CODE REVIEW

File: spring-cloud-netflix-core/src/test/java/org/springframework/cloud/netflix/feign/valid/FeignClientTests.java [hunk 1]
```diff
@@ -525,12 +564,16 @@ public void testParams() {
...
-	public void testHystrixCommand() {
+	public void testHystrixCommand() throws NoSuchMethodException {
 		HystrixCommand<List<Hello>> command = this.testClient.getHellosHystrix();
 		assertNotNull("command was null", command);
 		assertEquals(
 				"Hystrix command group name should match the name of the feign client",
 				"localapp", command.getCommandGroup().name());
+		String configKey = Feign.configKey(TestClient.class,
+			TestClient.class.getMethod("getHellosHystrix", (Class<?>[]) null));
+		assertEquals("Hystrix command key name should match the feing config key",
```
Comment (spencergibb) [inline spring-cloud-netflix-core/src/test/java/org/springframework/cloud/netflix/feign/valid/FeignClientTests.java]: typo: feing

File: spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/feign/FeignClientFactoryBean.java [hunk 2]
```diff
@@ -39,12 +39,15 @@
 import feign.codec.Decoder;
 import feign.codec.Encoder;
 import feign.codec.ErrorDecoder;
+import feign.hystrix.HystrixFeign;
```
Comment (spencergibb) [inline spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/feign/FeignClientFactoryBean.java]: This concerns me a little bit. Previously, this class only dealt with feign core classes. Now this introduces a dependency on the feign-hystrix module.

Comment (ErikKringen) [inline spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/feign/FeignClientFactoryBean.java]: Another option I was looking at would be to put it in `HystrixTargeter`, so the check to see if it's a HystrixBuilder is already done

Comment (spencergibb) [inline spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/feign/FeignClientFactoryBean.java]: Would you be willing to give that a try?

Comment (ErikKringen) [inline spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/feign/FeignClientFactoryBean.java]: I'll give it a go!

**Comment by ErikKringen** [comment]
@spencergibb Implemented the SetterFactory within HystrixTargeter as discussed

**Pull request merged**